src/components/include/cos_kernel_api.h

```diff
@@ -107,5 +107,63 @@ int cos_hw_detach(hwcap_t hwc, hwid_t hwid);
 void *cos_hw_map(struct cos_compinfo *ci, hwcap_t hwc, paddr_t pa, unsigned int len);
 int cos_hw_cycles_per_usec(hwcap_t hwc);
 
+/* Default API */
+/* Capabilities for Async activation end point */
+struct cos_aep_info {
+	tcap_t    tc;
+	thdcap_t  thd;
+	arcvcap_t rcv;
+};
+
+/* Default Component information */
+struct cos_defci {
+	struct cos_compinfo ci;
+	struct cos_aep_info sched_aep;
+};
+
+/* current component's default component information */
+extern struct cos_defci curr_defci;
+
+/*
+ * cos_defci_init: initialize the current component's global cos_defci struct using the standard boot capabilities layout.
+ */
+void cos_defci_init(void);
+/*
+ * cos_defci_init_ext: initialize the current component's global cos_defci struct using the parameters passed.
+ */
+void cos_defci_init_ext(tcap_t sched_tc, thdcap_t sched_thd, arcvcap_t sched_rcv, pgtblcap_t pgtbl_cap, captblcap_t captbl_cap, compcap_t comp_cap, vaddr_t heap_ptr, capid_t cap_frontier);
+/*
+ * cos_compinfo_get: returns the cos_compinfo pointer that points to the cos_compinfo struct inside cos_defci.
+ */
+struct cos_compinfo *cos_compinfo_get(void);
+/*
+ * cos_sched_aep_get: returns the AEP pointer that points to the current component's scheduler.
+ */
+struct cos_aep_info *cos_sched_aep_get(void);
```

Comment:
If you're using the `curr_` to describe this component's structures, I'd think you'd want to follow that through.  It might be the case that it would better to have a function to get the current `defci`, and separate functions to get the `aep` and the `compinfo` from *any* `defci`.  These functions feel very specific and it might be better to do something more general here.